### DISCUSSION TITLE
This project is rudderless

### DISCUSSION THREAD

**Discussion by knightmeister** [microsoft/microsoft-ui-xaml#8712] [General]
### Describe the bug

I'm raising this as a bug for visibility.

# This project is rudderless

I have just received multiple email notifications of issues closed, and looking at the number of closed items there is like over 900 closed since I last looked at the list yesterday. 

Management must love seeing that burndown chart!

But what about the people who are actually trying to use this framework? The framework is riddled with simple, trivial bugs and missing features. The framework is at least 12 years old (it was in the Win8 DP) - aren't Microsoft embarrassed by how much the mobile platforms have iterated in this time and how poorly WinUI compares?

I've invested significant amounts of my time into this project, raising issues, providing feedback and following it along. It started with great promise but it feels like there are 2 developers and an intern doing the development.

The fact Microsoft allowed the issue list to get so out of hand, with no effort to triage or respond is an embarrassment to the team and demonstrates a complete lack of care or interest for the community. At least in the beginning the team made an effort to triage and assign cases. How do you think new developers feel coming here, asking for help and getting nothing?

Those issues that have been closed, how much valuable feedback has the team lost because they didn't assign a developer to validate and investigate?

There has been no update to the timelines or feature map in ages. 

Why is there no consideration paid to developers building real apps (as opposed to things like sound boards or whoopie cushions) who require basic features like input validation? Look at the number of interactions with #179 and yet something so trivial has been sitting around since the start of 2019?

Further, using WinUI from managed code - which I'd wager 99% of devs do - is not a good experience. Why hasn't there been investment in better tooling? Anything that goes wrong you get a nonsense stack trace and no information with no assistance to determine the root cause. Don't even get me started on the lack of a designer.

I think the management team here needs to take a good look at how they communicate and interact with the community. I hope to be pleasantly surprised and that things will change, but after so many years of inaction I'm not going to hold my breath. It's a real shame because this framework genuinely had promise, yet it's been super neglected for years at this point.

To summarise, my advice:

1. Communicate openly with the community
2. In the interests of transparency, let the community know how many people are working on this product
3. Provide an up-to-date roadmap 
4. Participate in issues as they're raised

Or alternatively, just close this repository and go back to development like it's 1993. Perhaps we can then load bugs using PSS and wait for hotfixes...

### Steps to reproduce the bug

Use WinUI!

### Expected behavior

A UI framework that doesn't suck

### NuGet package version

None

**Comment by tdh1985** [comment]
Agreed.

**Comment by JackScottAU** [comment]
Absolutely. Communication is key in all projects, and even more important in open-source infrastructure like this!

Microsoft, we're not expecting perfection. But just tell us what we can expect!

**Comment by ghostidentity** [comment]
You can simply open the closed issue if you think the bug is still exist ? The thing is,  there is a possibility that the issues have been resolved already and if they spend some time to investigate thousands of old issues, it might not be efficient.

**Comment by huoyaoyuan** [comment]
They claimed to be open sourced for the fundamental components, but it doesn't happen for 3 years. Some trivial bugs can be fixed by people actually using them.

**Comment by dotMorten** [comment]
This has been communicated on the community calls and in the announcements, including a status update a couple days ago:
https://github.com/microsoft/microsoft-ui-xaml/discussions/8638
Every single issue that was closed was notified about it, and you can reopen your issues still.
Personally half the issues I’ve logged over the years and got notified about was no longer an issue or not applicable. I’m guessing this goes for quite a lot of issues.

I’m glad to see some action is being taken to rein this project in. I only see this as the project finally getting a rudder as opposed to how it has been run the past many years. Honestly this is all a good thing in my mind.

Let’s all help get rid of all the non issues (this one included that should have been a discussion) and focus on active bugs.

**Comment by knightmeister** [comment]
@dotMorten 
>Every single issue that was closed was notified about it, and you can reopen your issues still.

I'm aware of this. Numerous people have raised issues with it, but this is only a part of why I raised this issue. This project does **not** feel healthy. If your issues were no longer issues, or applicable, then you probably should have closed them. How many of the bulk closing of like 1k issues today were still issues and reasonable feature requests?

>I’m glad to see some action is being taken to rein this project in. 

Proves my point really, this project is rudderless. No effort has been put into triaging issues in a very long time. Development is not happening in an open manner, and the bulk closure of issues is not steering the ship (i.e. giving it a rudder), rather throwing ballast overboard in an effort to stop it sinking just a little longer.

**Comment by thatsofia** [comment]
> You can simply open the closed issue if you think the bug is still exist ? The thing is, there is a possibility that the issues have been resolved already and if they spend some time to investigate thousands of old issues, it might not be efficient.

No, you cannot reopen issues once they've been closed by a repo collaborator or bot. Them spending time to triage issues is what they should be doing. They could spend two days just cleaning up the repo and commenting and it'd be over and done with. The state of this repo and of WinUI shows that we really don't know what they're doing. Closing issues that users have created shows that they don't respect our time and effort. And it's not like they're continuing to make changes despite that, unlike some other projects. Our feedback is valuable. We don't get paid to update the issues we create. We get paid to build apps and we're going the extra mile to try and get a better experience. And just think of FOSS! They don't even get paid. 

We were excited for #179 where I worked at but they're now rebuilding with a web front-end. I think they felt tired of Microsoft's native UI frameworks too. And of course "cloud and mobile are the future"

And despite Microsoft just automatically closing old issues that are only "stale" because they never looked at them, they haven't decided to triage any recent ones. Some spammy issues were made a week ago and have not been closed. 

I'm very curious to see how many people work on WinUI and WASDK. I don't follow it very closely anymore but it's always felt like a side project by two people. Judging by the state of Windows, they've seemingly had difficulties getting in touch with designers across teams

**Comment by knightmeister** [comment]
The irony that the team actually triaged this issue and converted it to a discussion is not lost on me

**Reply by thatsofia** [reply]
And #8683 is still open while #2618 just got closed! Weird that they'd only move this to a discussion and not deal with the spam I had mentioned.

**Comment by pjmlp** [comment]
Indeed, here is another proof how ruderless it happens to be.

Apparently C++/WinRT, once upon a time sold to us as sucessor for C++/CX, and still being pushed as way to program WinUI applications in C++, has been placed in maintainance mode.

This poped up in a discussion thread a couple of months ago, on their repo, https://github.com/microsoft/cppwinrt/issues/1289?notification_referrer_id=NT_kwDOAAlKbLE1OTM0NjU0NjU4OjYwODg3Ng#issuecomment-1481303844

So effectively, there is no C++ tooling in development to support WinUI development scenarios, and nothing was communicated to the community in that regard, after forcing developers to waste tons of money in porting activities.

**Reply by llothar** [reply]
It's just the interface/language binding part and i do agree that can be frozen without problems until there is some real progress in the C++ world, like Herb Sutters new frontend language. 

What we need is documentation and features and documentation and examples and documentation.
Seriously how would you get any student to learn and understand the current state of C++ windows programming. Including all the harder infrastructure topics like building and deploying.
